spatialdata/_dl/datasets.py

```diff
@@ -0,0 +1,130 @@
+from typing import Callable, Optional
+
+import numpy as np
+from geopandas import GeoDataFrame
+from multiscale_spatial_image import MultiscaleSpatialImage
+from spatial_image import SpatialImage
+from torch.utils.data import Dataset
```

Comment:
I would make torch as optional depedency, therefore I think in __Init__ of this module or where the ImageTilesDataset is import, something like this would be needed

```python
try:
    from spatialdata._dl.datasets import ImageTilesDataset
except ImportError as e:
    _error: str | None = str(e)
else:
    _error = None
```

Reply:
would you add `torch` somewhere in `pyproject.toml` or it would be responsibility of the user to install it properly? I would go for the second.